### ISSUE TITLE
Regression of Bugzilla#27779 Xamarin.Forms.ReadOnlyListAdapter.IndexOf throws NotImplementedException

### ISSUE THREAD

**Issue by Adam-Langley** [xamarin/Xamarin.Forms#6195]
See https://bugzilla.xamarin.com/show_bug.cgi?id=27779

All details are there.
I did originally fill this all out - then GitHub blanked out my report when I submitted it... 

[XamForms-CrashSelectingItem.zip](https://github.com/xamarin/Xamarin.Forms/files/3180769/XamForms-CrashSelectingItem.zip)

**Comment by kingces95** [comment]
Def crashes when I select. Unfortunately, the preview debugger may have an issue because it doesn't stop on the exception. We'll have a look.

**Comment by kingces95** [comment]
Well, I gave this the old college try. But still seems our debugger doesn't wanna stop on the crash on iOS <10 and above clicking on any of the items causes the app to _self background_. Wha?

**Comment by kingces95** [comment]
Hm. If I deploy to a 10+ device then I get this exception having to do with the database connection. 

![image](https://user-images.githubusercontent.com/4120386/61835953-b7a57480-ae19-11e9-9212-c21fedc6e695.png)

**Comment by kingces95** [comment]
@Adam-Langley I'm not able to capture the exception you're seeing on <10 devices. I see an odd backgrounding behavior.. The market share for <10 devices is low and falling fast so unless this is blocking because your app specifically targets <10 devices for some reason I'd rather focus on getting it working on >10 devices. For that, it'd help to have a reproduction w/o the database logic.

**Comment by Adam-Langley** [comment]
Hi @kingces95 I've just updated:
MacOS 10.14.16
Xcode 10.3 (10G8)
Visual Studio Mac 8.2 (1105)

The issue still manifests on iOS Simulator 12.4.

I've removed the EF code as it's not relevant.
I made some modifications to deal with the EF issue you were getting anyway - that was just the linker stripping away too much (known issue according to EF team).

I've reattached the simplified project - manifesting the same crash.

[XamForms-CrashSelectingItem-POCO.zip](https://github.com/xamarin/Xamarin.Forms/files/3433345/XamForms-CrashSelectingItem-POCO.zip)

**Comment by kingces95** [comment]
Huh, I'm still not able to see the exception -- instead the app backgrounds itself.

**Comment by Adam-Langley** [comment]
Hi @kingces95  - what are you getting in Console? Is the app really backgrounding or crashing? I'm not aware of any capabilities of apps to 'background themselves' on iOS....

**Comment by kingces95** [comment]
Yeah, neither did I. It's weird. The appears to background itself but when its brought back forward it's recreated itself yet the debugger hasn't detached. So, it's just weird...

[bug.zip](https://github.com/xamarin/Xamarin.Forms/files/3688333/bug.zip)

**Comment by Adam-Langley** [comment]
Hi @kingces95 - I see - the app isn't backgrounded, it's crashed. When an app crashes on iOS, it just disappears. The multitasking window will still show the last captured screenshot however - leading you to think it's still running/backgrounded - but clicking it just relaunches the app. 
I don't know why your debugger doesn't catch it however. Can you post your Console logs? I expect there will be something in there regardless of what happened with the debugger.

**Comment by kingces95** [comment]
Yeah, I'm not seeing anything in the VS console output logs and the debugger is not capturing the exception. I guess we'll pass this over to the iOS team. I'm running the latest internal preview VS as of this post.

**Comment by Adam-Langley** [comment]
Hi @kingces95, sorry I wasn't explicit enough - I'm asking about the OSX Console, not the debugger output in Visual Studio. We need to take VS out of the equation if you're not seeing any exception messages.
Using Spotlight, and search for "Console".

![image](https://user-images.githubusercontent.com/16373742/66535259-8a917400-eb75-11e9-9957-ae1baea66847.png)

**Comment by kingces95** [comment]
VS is supposed to pipe that console. Can you paste a screenshot of what you're seeing so I know what I'm looking for?

**Comment by kingces95** [comment]
Opened an issue for the VS team: https://developercommunity.visualstudio.com/content/problem/772049/expected-vs-to-break-when-ios-app-crashes.html

**Comment by Adam-Langley** [comment]
Hi @kingces95, I'm no longer getting the NotImplementedException - but now I'm getting the same as you - immediate crash to springboard.
I've updated my XCode to 11.1, and am also now on Catalina - so that could be why - were you running a preview of any of these builds when you first ran the tests?

The crash happens when I select any item in the list.

This is what I'm getting if I attach XCode to the Xamarin app process - XCode has no trouble catching the exception - is obviously some bad memory state however:

![image](https://user-images.githubusercontent.com/16373742/66685424-fa6c3f80-ecd8-11e9-94ce-8e14fb5348cd.png)

**Comment by kingces95** [comment]
Hm. Well, that could be an access violation inside of the runtime which might explain why the debugger isn't catching the managed exception. Soooooo... this seems like a mono issue. I'd cross post this issue there with a link back here and save this issue for the NYI exception so we can fix that once mono fixes this issue.
